### DISCUSSION TITLE
DataTemplates in MenuFlyout?

### DISCUSSION THREAD

**Discussion by sachssoft** [AvaloniaUI/Avalonia#15380] [Q&A]
Where is the DataTemplates in MenuFlyout?

I need it for different view models.

Any ideas?

**Comment by timunie** [comment]
Use ItemTemplate. 3 options for different DataTypes:
- FuncDataTemplate: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/FuncDataTemplateSample
- Implment IDataTemplate: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/IDataTemplateSample
- Add a ContentControl as DataTemplate which has the DataTemplates-section as needed.
